sympy/functions/special/holonomic.py

```diff
@@ -0,0 +1,161 @@
+    holonomic(-Dx**2*x - Dx + 1/x)
+
+    For details see ore_algebra package in Sage
+    """
+
+    def __init__(self, ann, x, *args):
+        if len(args) is 2:
+            self.cond = args[0]
+            self.cond_point = args[1]
+        elif len(args) is 1:
+            self.cond = args[0]
+            self.cond_point = 0
+        self.ann = ann
+        Dx = symbols('Dx')
+        self.gen = Dx
+        self.var = x
+
+        
+    def __repr__(self):
+        return 'holonomic(%s)' % sstr(self.ann)
+
+    def __add__(self, other):
+
+        deg1 = poly(self.ann, self.gen).degree()
+        deg2 = poly(other.ann, other.gen).degree()
+        dim = max(deg1, deg2)
+        rowsself = [self.ann]
+        rowsother = [other.ann]
+        gen = DifferentialOperator(self.gen)
+
+        # constructing annihilators up to order dim
+        for i in range(dim - deg1):
+            diff1 = (gen*DifferentialOperator(rowsself[-1])).eq
+            rowsself.append(diff1)
+
+        for i in range(dim - deg2):
+            diff2 = (gen*DifferentialOperator(rowsother[-1])).eq
+            rowsother.append(diff2)
+
+        row = rowsself + rowsother
+
+        # constructing the matrix of the ansatz
+        r = [[poly(expr, self.gen).nth(i) for i in xrange(dim + 1)] for expr in (row)]
+        from sympy.matrices import Matrix
+        r = Matrix(r).transpose()
+        homosys = [[0 for q in xrange(dim + 1)]]
+        homosys = Matrix(homosys).transpose()
+
+        # solving the linear system using gauss jordan solver
+        solcomp = r.gauss_jordan_solve(homosys)
+        sol = (r).gauss_jordan_solve(homosys)[0]
+
+        # if a solution is not obtained then increasing the order by 1 in each iteration
+        while sol.is_zero:
+            dim += 1
+            diff1 = (gen*DifferentialOperator(rowsself[-1])).eq
+            rowsself.append(diff1)
+            diff2 = (gen*DifferentialOperator(rowsother[-1])).eq
+            rowsother.append(diff2)
+            row = rowsself + rowsother
+            r = [[poly(expr, self.gen).nth(i) for i in xrange(dim + 1)] for expr in (row)]
+            r = Matrix(r).transpose()
+            homosys = [[0 for q in xrange(dim + 1)]]
+            homosys = Matrix(homosys).transpose()
+            solcomp = r.gauss_jordan_solve(homosys)
+            sol = r.gauss_jordan_solve(homosys)[0]
+
+        # removing the symbol if any from the solution
+        if sol.is_symbolic():
+            sol = solcomp[0]/solcomp[1]
+
+        # taking only the coefficients needed to multiply with `self`
+        # can be also be done the other way by taking R.H.S and multiply with `other`
+        sol = sol[:dim + 1 - deg1]
+
+        #construct expression from the coefficients
+        expr = 0
+        for i, j in enumerate(sol):
+            expr += ((self.gen)**i)*j
+
+        return(HoloFunc((DifferentialOperator(expr)*DifferentialOperator(self.ann)), self.var))
```

Comment:
Don't put parentheses next to `return`. It should just be `return HoloFunc((DifferentialOperator(expr)*DifferentialOperator(self.ann)), self.var)`